src/utils.ts

```diff
@@ -111,5 +111,9 @@ export function getHkdf(ck: bytes32, ikm: bytes): Hkdf {
 }
 
 export function isValidPublicKey(pk: bytes): boolean {
-  return x25519.publicKeyVerify(pk.slice(0, 32));
+  if(pk.length !== 32 || pk.equals(Buffer.alloc(32))){
+    return false;
+  }
+
+  return true;
```

Comment:
@morrigan Any idea how to implement proper public key verification?

Reply:
Can't find anything. I guess no big deal as it won't pass DH otherwise